src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -252,58 +253,131 @@ BuildSelectStatement(Query *masterQuery, List *masterTargetList, CustomScan *rem
 		topLevelPlan = (Plan *) limitPlan;
 	}
 
-	/* (6) finally set our top level plan in the plan tree */
-	selectStatement->planTree = topLevelPlan;
+	/* (6) set top level plan in the plantree and copy over some things from
+	 * PlannerInfo */
+	FinalizeStatement(root, selectStatement, topLevelPlan);
+
+	foreach(targetEntryCell, masterTargetList)
+	{
+		TargetEntry *targetEntry = lfirst(targetEntryCell);
+		columnNameList = lappend(columnNameList, makeString(targetEntry->resname));
+	}
+
+	customScanRangeTableEntry = RemoteScanRangeTableEntry(columnNameList);
+	list_head(selectStatement->rtable)->data.ptr_value = customScanRangeTableEntry;
```

Comment:
Can there be more stuff in the rtable here? If so, ignore. If not, maybe `selectStatement->rtable = list_make1(customRangeTableEntry);`?